common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -173,26 +174,546 @@ public Collation(
     }
 
     /**
-     * Constructor with comparators that are inherited from the given collator.
+     * Collation ID is defined as 32-bit integer. We specify binary layouts for different classes of
+     * collations. Classes of collations are differentiated by most significant 3 bits (bit 31, 30
+     * and 29), bit 31 being most significant and bit 0 being least significant.
+     * ---
+     * General collation ID binary layout:
+     * bit 31:    1 for INDETERMINATE (requires all other bits to be 1 as well), 0 otherwise.
+     * bit 30:    0 for predefined, 1 for user-defined.
```

Comment:
do we support it now?

Reply:
We don't support user-defined collations for now. This is reserved for future use.